sed/calibrator/delay.py

```diff
@@ -25,22 +28,102 @@ class DelayCalibrator:
     def __init__(
         self,
         config: dict = None,
-    ):
+    ) -> None:
         """Initialization of the DelayCalibrator class passes the config.
 
         Args:
             config (dict, optional): Config dictionary. Defaults to None.
         """
-        if config is None:
-            config = {}
-
-        self._config = config
+        self._config: dict = config or {}
+        self.loader: str = self._config["core"]["loader"]
 
-        self.adc_column = self._config["dataframe"]["adc_column"]
-        self.delay_column = self._config["dataframe"]["delay_column"]
-        self.calibration: Dict[Any, Any] = {}
+        self.adc_column: str = self._config["dataframe"].get("adc_column", None)
+        self.delay_stage_column: str = self._config["dataframe"].get("delay_stage_column", None)
+        if self.delay_stage_column is None and self.adc_column is None:
+            raise ValueError("No delay stage column specified.")
+        self.delay_column: str = self._config["dataframe"]["delay_column"]
+        self.calibration: Dict[str, Any] = self._config["delay"].get("calibration", {})
+        self.fluctuations: Dict[str, Any] = self._config["delay"].get("fluctuations", {})
```

Comment:
I'd suggest to make a similar contruct here as for the energy shifts, to create a shift dictionary, with constant and column-dependent shifts.

Reply:
agreed!